packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -202,3 +211,57 @@ export async function processDRResponse({
     });
   }
 }
+
+export async function processDRResponseMTOM({
+  drResponse: { response, success, gateway, outboundRequest, contentType },
+  s3Utils,
+}: {
+  drResponse: DRSamlClientResponse;
+  s3Utils: S3Utils;
+}): Promise<OutboundDocumentRetrievalResp> {
+  if (!contentType) {
+    throw new Error("No content type found in response");
+  }
+  if (success === false) {
+    return handleHTTPErrorResponse({
+      httpError: response,
+      outboundRequest,
+      gateway,
+    });
+  }
+  try {
+    const documentResponses = parseMTOMResponse(response, contentType);
+    return await handleSuccessResponse({
+      documentResponses,
+      outboundRequest,
+      gateway,
+      s3Utils,
+    });
+  } catch (error) {
+    capture.error("Error parsing MTOM response", {
+      extra: {
+        error,
+        response,
+        contentType,
+      },
+    });
+    return handleErrorMTOMResponse({
+      outboundRequest,
+      gateway,
+    });
```

Comment:
For now with MTOM, all errors are handled in the try catch here and the error messages going into the DB are generic and don't actually have the error message from inside the SOAP response. 
Its part of the followup MTOM ticket to refactor the code so that we have the same detailed error handling for MTOM as regular DRs.